library/core/class.upload.php

```diff
@@ -318,13 +323,14 @@ public function saveAs($source, $target, $options = [], $copy = false) {
         if (!$handled) {
             $target = PATH_UPLOADS.'/'.$parsed['Name'];
             if (!file_exists(dirname($target))) {
-                mkdir(dirname($target));
+                mkdir(dirname($target), 0777, true);
```

Comment:
Files should be 775 and directories 664.
Same below.

Reply:
0777 is actually the default file mode for [`mkdir`](http://php.net/mkdir). Any directories it was making in the past would've also been 0777. The only thing I actually changed here allowing for the creation of nested directories.

Reply:
Let's create an issue about it

Reply:
https://github.com/vanilla/vanilla/issues/6254

Reply:
https://github.com/vanilla/vanilla/issues/6254